packages/inter-protocol/src/vpool-xyk-amm/pool.js

```diff
@@ -92,7 +90,14 @@ export const definePoolKind = (baggage, ammPowers, storageNode, marshaller) => {
    * @returns {ImmutableState & MutableState}
    */
   const poolInit = (liquidityZcfMint, poolSeat, secondaryBrand) => {
-    /** @type {PublishKit<{secondary: Amount, central: Amount}>} */
+    /** @type {StoredPublishKit<InitPublication>} */
+    const { publisher: initPublisher } = makeStoredPublishKit(
+      E(storageNode).getChildNode('init'),
+      marshaller,
+    );
+    initPublisher.finish({
+      liquidityIssuerRecord: liquidityZcfMint.getIssuerRecord(),
```

Comment:
We need the issuer's id. So `E(board).getId(liquidityZcfMint.getIssuerRecord().issuer)` effectively I think

Reply:
Hmm, we don't have a board here. With this way that gives you the whole issuer record, can you look up the ID in the board?

Reply:
There's no way for the client to look up the id in the board with rpc, we run into the same issue where that takes a chain transaction (and we don't even fully have a way to do such chain transactions yet).

Reply:
I assume we can pass the board through from the contract, but perhaps that's too messy, which I was afraid would be the case.

Reply:
@samsiegart the dapp gets the issuer the same way it gets brands: by deserializing using the `importContext`.

Reply:
Ah... objects deserialized with `importContext` are `SEVERED`; i.e. you can't actually call any methods on them. And we need an issuer that supports `E(issuer).getEmptyPurse()`. I think you have this working with a hard-coded board-id, right @samsiegart ? You can get the appropriate boardId by manually parsing the capData published in the storage node.

@turadg do we have an example of what `agoric follow` or `agd query vstorage ...` outputs for this `.init` node?

Reply:
The `suggestIssuer` method requires the issuer id specifically, not the issuer itself. The user needs to be able to verify the exact id before they give it a petname for their wallet.

Reply:
And yes I have it working with a hardcoded boardId. The wallet contract handles looking up the issuer by its id and calling `E(issuer).getEmptyPurse()`

Reply:
> The suggestIssuer method requires the issuer id specifically, not the issuer itself.

got it... then a trivial `x => x` should work for `convertSlotToVal`.

Reply:
@samsiegart I just checked out this branch and fired up the chain; see where the boardid for the LP issuer is?

```
$ ~/go/bin/agd query vstorage data published.amm.pool0.init -o json | jq '.value | fromjson | {body: .body | fromjson, slots}'
{
  "body": {
    "liquidityIssuerRecord": {
      "assetKind": "nat",
      "brand": {
        "@qclass": "slot",
        "iface": "Alleged: IbcATOMLiquidity brand",
        "index": 0
      },
      "displayInfo": {
        "assetKind": "nat",
        "decimalPlaces": 6
      },
      "issuer": {
        "@qclass": "slot",
        "iface": "Alleged: IbcATOMLiquidity issuer",
        "index": 1
      }
    }
  },
  "slots": [
    "board0188",
    "board02810"
  ]
}
```